web/frontend/src/components/slam.vue

```diff
@@ -58,56 +62,99 @@ const fetchSLAMMap = (name: string) => {
           code: status.code,
           metadata: status.metadata,
         };
-        displayError(error);
+        reject(error);
       }
     });
     getPointCloudMapStream.on('end', (end) => {
       if (end === undefined || end.code !== 0) {
         // the error will be logged in the 'status' callback
         return;
       }
-      pointcloud = concatArrayU8(chunks);
-      pointCloudUpdateCount += 1;
+      console.log('fetchSLAMMap end');
+      resolve(concatArrayU8(chunks));
+      console.log('fetchSLAMMap end finished');
     });
+    // });
   });
 };
 
-const fetchSLAMPose = (name: string) => {
-  return nextTick(() => {
+const fetchSLAMPose = (name: string): Promise<commonApi.Pose | undefined> => {
+  return new Promise((resolve, reject): void => {
+    console.log('fetchSLAMPose start');
     const req = new slamApi.GetPositionNewRequest();
     req.setName(name);
     props.client.slamService.getPositionNew(req, new grpc.Metadata(), (error, res): void => {
       if (error) {
-        displayError(error);
+        reject(error);
         return;
       }
-      pose = res!.getPose()!;
+      console.log('fetchSLAMPose end', res!.getPose()!);
+      resolve(res!.getPose()!);
+      console.log('fetchSLAMPose end finished');
     });
   });
 };
 
 const refresh2d = async (name: string) => {
```

Comment:
Same here, I'm not sure why refresh2d needs to be async, since we always call await on the result immediately.